lab_4_retrieval_w_clustering/main.py

```diff
@@ -113,6 +153,36 @@ def put_corpus(self, corpus: Corpus) -> None:
                 or if input arguments are empty,
                 or if methods used return None.
         """
+        if not isinstance(corpus, list) or not corpus:
+            raise ValueError('No argument or unacceptable type of argument')
+
+        tok_pars = []
+
+        for item in corpus:
+            tok_text = self._tokenizer.tokenize(item)
+            if tok_text is None or not isinstance(tok_text, list):
+                raise ValueError('Tokenized text is None or not isinstance list')
+
+            if len(tok_text):
+                tok_pars.append(tok_text)
+                self.__documents.append(item)
+
+        self._vectorizer.set_tokenized_corpus(tok_pars)
+        self._vectorizer.build()
+
+        vect_pars = []
+
+        for text in tok_pars:
+            vector = self._vectorizer.vectorize(text)
+            if not isinstance(vector, tuple) or not vector:
+                raise ValueError('No vector or vector not isinstance tuple')
+
+            vect_pars.append(vector)
```

Comment:
can you do this directly in `self.__vectors__` and not create a separate variable?